src/immutable/FetchableData.ts

```diff
@@ -0,0 +1,27 @@
+import { $ReadOnly } from 'utility-types';
+import { Record } from 'immutable';
+
+import { DefaultValues } from '~types/index';
+
+interface Shared<R> {
+  record: R;
+  error?: string | void;
+  isFetching?: boolean;
+  lastFetchedAt?: Date;
+}
+
+export type FetchableDataType<R> = $ReadOnly<Shared<R>>;
```

Comment:
FYI there is a `Readonly` built-in TS type, so we could replace `$ReadOnly` if we wanted to 🤷‍♂

Reply:
I replaced all of them!